cmd/metal-api/internal/metal/provisioning.go

```diff
@@ -4,9 +4,56 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/looplab/fsm"
 	"go.uber.org/zap"
 )
 
+// FSM states
+const (
+	FsmStateInitial          string = "Initial State"
+	FsmStatePXEBooting       string = "PXE Booting state"
+	FsmStatePreparing        string = "Rreparing state"
+	FsmStateRegistering      string = "Registering state"
+	FsmStateWaiting          string = "Waiting state"
+	FsmStateInstalling       string = "Installing state"
+	FsmStateBootingNewKernel string = "Booting New Kernel state"
+	FsmStateProvisioned      string = "Provisioned state"
+	FsmStatePlannedReboot    string = "Planned Reboot state"
+	FsmStateCrashed          string = "Crashed state"
+	FsmStateAlive            string = "Alive state"
+	FsmStateResestFailCount  string = "Reset Fail Count state"
+)
+
+var allStates = []string{
+	FsmStateInitial,
+	FsmStatePXEBooting,
+	FsmStatePreparing,
+	FsmStateRegistering,
+	FsmStateWaiting,
+	FsmStateInstalling,
+	FsmStateBootingNewKernel,
+	FsmStateProvisioned,
+	FsmStatePlannedReboot,
+	FsmStateCrashed,
+	FsmStateAlive,
+	FsmStateResestFailCount,
+}
+
+var events = []fsm.EventDesc{
+	{Name: string(ProvisioningEventPXEBooting), Src: []string{FsmStateInitial, FsmStatePlannedReboot, FsmStatePXEBooting, FsmStateCrashed, FsmStateProvisioned, FsmStateResestFailCount}, Dst: FsmStatePXEBooting},
+	{Name: string(ProvisioningEventPreparing), Src: []string{FsmStatePXEBooting, FsmStateInitial, FsmStatePlannedReboot, FsmStateCrashed, FsmStateProvisioned, FsmStateResestFailCount}, Dst: FsmStatePreparing},
+	{Name: string(ProvisioningEventRegistering), Src: []string{FsmStatePreparing, FsmStateInitial, FsmStateResestFailCount}, Dst: FsmStateRegistering},
+	{Name: string(ProvisioningEventWaiting), Src: []string{FsmStateRegistering, FsmStateInitial, FsmStateResestFailCount}, Dst: FsmStateWaiting},
+	{Name: string(ProvisioningEventInstalling), Src: []string{FsmStateWaiting, FsmStateInitial, FsmStateResestFailCount}, Dst: FsmStateInstalling},
+	{Name: string(ProvisioningEventBootingNewKernel), Src: []string{FsmStateInstalling, FsmStateInitial, FsmStateResestFailCount}, Dst: FsmStateBootingNewKernel},
+	{Name: string(ProvisioningEventPhonedHome), Src: []string{FsmStateBootingNewKernel, FsmStateProvisioned, FsmStateInitial, FsmStateResestFailCount}, Dst: FsmStateProvisioned},
+	{Name: string(ProvisioningEventPlannedReboot), Src: allStates, Dst: FsmStatePlannedReboot},
+	{Name: string(ProvisioningEventCrashed), Src: []string{FsmStateInitial}, Dst: FsmStateCrashed},
+	{Name: string(ProvisioningEventResetFailCount), Src: allStates, Dst: FsmStateResestFailCount},
+}
+
+var callbacks = map[string]fsm.Callback{}
```

Comment:
I guess these will be implemented at a later stage? If so, you can imply your intent with `// TODO: to be implemented` or something like this.